lib/wibox/container/place.lua

```diff
@@ -0,0 +1,187 @@
+---------------------------------------------------------------------------
+-- A container used to place smaller widgets into larger space.
+--
+--@DOC_wibox_container_defaults_place_EXAMPLE@
+-- @author Emmanuel Lepage Vallee &lt;elv1313@gmail.com&gt;
+-- @copyright 2016 Emmanuel Lepage Vallee
+-- @release @AWESOME_VERSION@
+-- @classmod wibox.container.place
+---------------------------------------------------------------------------
+
+local setmetatable = setmetatable
+local base = require("wibox.widget.base")
+local util = require("awful.util")
+
+local place = { mt = {} }
+
+-- Take the widget width/height and compute the position from the full
+-- width/height
+local align_fct = {
+    left   = function(_  , _   ) return 0                         end,
+    center = function(wdg, orig) return math.max(0, (orig-wdg)/2) end,
+    right  = function(wdg, orig) return math.max(0, orig-wdg    ) end,
```

Comment:
I'd argue that the `math.max` calls here are wrong: The child should be centered/right-aligned even when not enough space is available.